ts/Series/SeriesOnPointComposition.ts

```diff
@@ -0,0 +1,517 @@
+                    d: SVGRenderer.prototype.crispLine([
+                        ['M', xFrom, yFrom],
+                        ['L', xTo, yTo]
+                    ], width, 'ceil')
+                };
+
+                attribs['stroke-width'] = width;
+
+                if (!chart.styledMode) {
+                    attribs.stroke = color;
+                    attribs.dashstyle = dashStyle;
+                }
+
+                return attribs;
+            }
+        }
+    }
+
+    /**
+     * Draw connector line that connects series and initial point's position.
+     * @private
+     */
+    function seriesDrawConnector(this: Series): void {
+        if (this.onPoint) {
+            if (!this.onPoint.connector) {
+                (this.onPoint.connector as any) = this.chart.renderer.path()
+                    .addClass('highcharts-connector-seriespoint')
+                    .attr({
+                        zIndex: -1
+                    })
+                    .add(this.markerGroup);
+            }
+
+            const attribs = this.getConnectorAttributes();
+            if (this.onPoint.connector) {
+                (this.onPoint.connector as any)['attr'](attribs);
+            }
+        }
+    }
+
+    function seriesDrawPoints(this: Series): void {
+        (drawPointsFunctions as any)[this.type + 'DrawPoints'].call(this);
+
+        this.seriesDrawConnector();
+    }
+
+    /* *
+     *
+     *  Classes
+     *
+     * */
+
+    /**
+     * @private
+     */
+    export class Additions {
+
+        /* *
+         *
+         *  Constructors
+         *
+         * */
+
+        /**
+         * @private
+         */
+        public constructor(series: SeriesComposition) {
+            this.series = series;
+            this.options = series.options.onPoint;
+        }
+
+        /* *
+         *
+         *  Properties
+         *
+         * */
+
+        public series: SeriesComposition;
+
+        public connector?: SVGElement = void 0 as any;
+
+        public options?: OnPoint;
+    }
+
+}
+
+/* *
+ *
+ *  Default Export
+ *
+ * */
+
+export default SeriesOnPointComposition;
+
+/* *
+ *
+ *  API Options
+ *
+ * */
+
+/**
+ * Color of a connector line. By default it's the series' color.
+ *
+ * @requires    modules/series-on-point
+ * @since       next
+ * @type        {string}
+ * @apioption   plotOptions.series.onPoint.connectorOptions.color
+ */
+
+/**
+ * A name for the dash style to use for connector.
+ *
+ * @requires    modules/series-on-point
+ * @since       next
+ * @type        string
+ * @apioption   plotOptions.series.onPoint.connectorOptions.dashStyle
+ */
+
+/**
+ * Pixel width of a connector line.
+ *
+ * @default     1
+ * @requires    modules/series-on-point
+ * @type        {number}
+ * @since       next
+ * @product     highcharts highstock
+ * @apioption   plotOptions.series.onPoint.connectorOptions.width
+*/
+
+/**
+ * An id of the point that we connect series to.
```

Comment:
Should the series type limitation be mentioned here? There are some series on which points we cannot (hopefully: yet) add series, like pie, 3d?, indicators?, etc.